### PR TITLE
[cfgfile ]Add new port

### PR DESCRIPTION
**Pull request by igormironchik**
**Describe the pull request**
  Add cfgfile library version 0.2.8.2.1.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, No.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/cfgfile/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+vcpkg_install_cmake()
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake TARGET_PATH share/cfgfile)
+
+vcpkg_copy_tools(TOOL_NAMES cfgfile.generator)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin")
+
+file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/cfgfile RENAME copyright)
```
Comment (JonLiu1993) [inline ports/cfgfile/portfile.cmake:29]: @igormironchik ,Please change"share/cfgfile" to "share/{PORT}"

File: ports/cfgfile/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "cfgfile",
+  "version-string": "0.2.8.2.1",
```
Comment (JonLiu1993) [inline ports/cfgfile/vcpkg.json:3]: Please change "version-string" to "version"

File: versions/c-/cfgfile.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "55a4bf4710b8f0a921921635fbb4411bcb6347ea",
+      "version-string": "0.2.8.2.1",
```
Comment (JonLiu1993) [inline versions/c-/cfgfile.json:5]: Please change "version-string" to "version"

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: versions/c-/cfgfile.json (hunk 3 above)
Comment (igormironchik) [reply versions/c-/cfgfile.json:5]: Could you tell me, please, do I need to change `git-tree` here if I will make new commit?

**Comment by igormironchik** [comment]
@JonLiu1993 How can I rerun checks? Failing is because of problems on servers...

File: ports/cfgfile/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    URLS "https://github.com/igormironchik/cfgfile/releases/download/0.2.8.2.1/cfgfile-0.2.8.2.1-full.tar.gz"
+    FILENAME "cfgfile02821.tar.gz"
+    SHA512 b51bed3e83673dcab281c90c8df7a14746ed475113327a84a20e9fa474a9aad79fd90cac7d8eb467d94881921b3a3a9b33d2bf793da975d64e3d52acca153ed3
+)
+
+vcpkg_extract_source_archive_ex(
+    OUT_SOURCE_PATH SOURCE_PATH
+    ARCHIVE ${ARCHIVE}
+)
+
```
Comment (autoantwort) [inline ports/cfgfile/portfile.cmake:13]: Maybe prefer [vcpkg_from_github](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_from_github.md)

Comment (igormironchik) [reply ports/cfgfile/portfile.cmake:13]: No, not for this library. This library can't be set up with `vcpkg_from_github` as internal tool (`cfgfile.generator`) uses my own library as submodule. I didn't find a way to set up submodules in `vcpkg`. I prepared special archive specially for C++ package managers like `vcpkg` and `conan`.

Maybe I missed something, so could you point me out please if so?

**Pull request closed without merging**